plugins/module_utils/network/sonic/config/route_maps/route_maps.py

```diff
@@ -2159,6 +2194,38 @@ def get_delete_route_map_replaced_set_groupings(self, command, cmd_rmap_have,
                     }
                     dict_delete_requests.append(request)
 
+            # Check for deletion of ip_next_hop attributes.
+            # As an optimization, avoid deleting attributes that will be replaced
+            # by the received command.
+            ip_next_hop_deleted_members = {}
+            if 'ip_next_hop' not in cfg_set_top:
+                if command['set'].get('ip_next_hop'):
+                    command['set'].pop('ip_next_hop')
+                    if command['set'] is None:
+                        command.pop('set')
+                    return
+
+            else:
+                # Delete eligible configured ip_next_hop members.
+                cfg_ip_next_hop_key_set = set(cfg_set_top['ip_next_hop'].keys())
+                cmd_ip_next_hop_key_set = ([])
+                if cmd_set_top.get('ip_next_hop'):
+                    cmd_ip_next_hop_key_set = set(cmd_set_top['ip_next_hop'].keys())
+
+                set_uri = set_uri_attr['ip_next_hop']
+                for ip_next_hop_key in cfg_ip_next_hop_key_set.difference(cmd_ip_next_hop_key_set):
+                    ip_next_hop_deleted_members[ip_next_hop_key] = \
+                        cfg_set_top['ip_next_hop'][ip_next_hop_key]
+                    request = {'path': set_uri[ip_next_hop_key], 'method': DELETE}
+                    dict_delete_requests.append(request)
+
+                if ip_next_hop_deleted_members:
+                    # Update the list of deleted ip_next_hop attributes in the "command" dict.
+                    if not cmd_set_top.get('ip_next_hop'):
+                        command['set']['ip_next_hop'] = {}
+
+                command['set']['ip_next_hop'] = ip_next_hop_deleted_members
```

Comment:
This should only be done when the "if ip_next_hop_deleted_member" condition is true, so it should be indented to place it in the 'if' block.

```suggestion
                    command['set']['ip_next_hop'] = ip_next_hop_deleted_members
```

Reply:
Updated.

Reply:
Verified.